daemon/container.go

```diff
@@ -329,6 +329,11 @@ func (daemon *Daemon) verifyContainerSettings(platform string, hostConfig *conta
 		return nil, errors.Errorf("invalid restart policy '%s'", p.Name)
 	}
 
+	// validate isolation mode
+	if !hostConfig.Isolation.IsValid() {
+		return nil, errors.Errorf("invalid isolation '%s' on this daemon", hostConfig.Isolation)
```

Comment:
"on this daemon" may not be so informative (the error returned by the client will be something like:

```
Error returned by daemon: invalid isolation 'foobar' on this daemon
```

Perhaps include the platform in the error message, e.g.:

```go
return nil, errors.Errorf("isolation mode '%s' is not supported for %s containers", hostConfig.Isolation, platform)
```
Which would show something like:

```
Error returned by daemon: isolation mode 'foobar' is not supported for linux containers
```

(but better suggestions are welcome)

Reply:
I think it is more runtime.GOOS that makes sense there (maybe later on, Microsoft will be capable of running linux containers with process isolation leveraging WSL).
LCOW specific validation is done when creating the container itself (and I don't want to add yet another LCOW settings validation here...)